### ISSUE TITLE
keyboard shortcuts

### ISSUE THREAD

**Issue by danielroe** [elk-zone/elk#39]
like default mastodon ui + twitter, we can implement shortcuts, e.g. `g - n` goes to notifications, `g - h` to home, etc.

**Comment by forestsystem** [comment]
Thank you for working on this, Hartmut!

I would love to have the option to use keyboard shortcuts. If possible, maybe Elk could improve on Mastodon's functionality, unless there's a reason not to: Mastodon's "j"/"k" keyboard shortcuts to move up and down tweets are pretty much unusable for me. The selected post goes at the bottom of the screen, partially but not completely hidden. This means I only see the bottom of the post above it if it's long. And I use this feature on Tumblr and Twitter mostly to skim more quickly, so it's super unhelpful to have posts jumping around, instead of the top of each post showing up in the same place. :(

It would be really cool for me if Elk did Tumblr/Twitter's version, instead of Mastodon's version!

**Comment by AmeliaBR** [comment]
> If possible, maybe Elk could improve on Mastodon's functionality

Agreed. The current Mastodon web implementation of keyboard navigation is very broken & should not be used as a template:
- scroll position is usually wrong (as @forestsystem explained),
- hijacking of arrow keys means you cannot tweak scroll position or scroll long posts without skipping to the next one,
- the shortcuts only work if you first get focus into the main feed,
- expanding conversations or similar actions lose the current place in the feed when you come back to it.

Last I checked, Twitter was pretty good at this (but who knows, it may have broken since). Facebook as well (except that it's easy to end up with focus in a reply box and type "jjj" trying to scroll!).

I would also suggest that you plan for customizable shortcuts from the beginning (i.e., use a dynamic internal map to connect keys with action), so that you can support other languages & keyboard layouts.

You'll also want to work with some screen reader devs to make sure the custom keyboard handling is correctly communicated to the screen reader (which have their own shortcuts).

**Comment by boehs** [comment]
Recieved request for `z` for zen mode

**Comment by leedo** [comment]
I'm a huge fan of Elk and also keyboard shortcuts :) It would be awesome to have `j` / `k` for navigating to next / prev message. I understand these may not be intuitive choices for most, but it seems fairly common in microblogging platforms.

**Comment by hartmut-co-uk** [comment]
Hi everyone, I'm planning to pick this up again.
`j` / `k` navigation actually is a big challenge to make it work with vue-virtual-scroller.

I'll add `z` for zen mode to the list.

Please refer to https://github.com/elk-zone/elk/pull/319 for technical discussions.
